cvmfs/catalog.h

```diff
@@ -163,7 +163,7 @@ class Catalog : public SingleCopy {
   void TakeDatabaseFileOwnership();
   void DropDatabaseFileOwnership();
   bool OwnsDatabaseFile() const {
-    return NULL != database_ && database_->OwnsFile();
+    return ((database_ != NULL) && database_->OwnsFile()) || managed_database_;
```

Comment:
Unfortunately not, I don't like it either. But a `Catalog` object can exist without a database object attached to it (yet). Therefore a `Catalog` needs to hold this state in order to forward it to its `CatalogDatabase` as soon as it gets attached to it.